### PR TITLE
type load

### PR DESCRIPTION
**Pull request by RCopeland**

### CODE REVIEW

File: src/patterns/fundamentals/typography/fallback.hbs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+name: Fallback Fonts
+order: 11
+doclayout: true
+---
+<div class="drizzle-o-ContentGrouping">
+    <p class="drizzle-b-P">
+      In the event that a client has issues loading typefaces from typekit, we've defined the
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/fallback.hbs]: Since we arent loading from typekit, we should remove that here

**Comment by RVMendoza** [comment]
related to issue https://github.com/sparkdesignsystem/spark-design-system/issues/666

image

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+name: Setting Up
+order: 1
+doclayout: true
+---
+<p>In order to use a webfont with Spark, several variables will need to be set in your app's sass build. These should be
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
<p>In order to use a webfont with Spark, several variables will need to be set in your app's Sass build. These should be
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  set as follows, before importing spark's scss.</p>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
  set as follows, before importing Spark's SCSS.</p>
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+<table class="drizzle-c-Table drizzle-c-Table--wide sprk-u-mbm">
+  <colgroup>
+    <col class="sprk-u-Width-40">
+    <col>
+  </colgroup>
+  <thead>
+  <tr>
+    <th>Sass Variable</th>
+    <th>type</th>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
    <th>Type</th>
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  <tr>
+    <td class="sprk-u-FontWeight--bold">$heading-font-name</td>
+    <td>string</td>
+    <td>This tells Spark the name of the font that will be loaded for display level type and is the name that will be
+    placed first in the heading font stack.</td>
+  </tr>
+  <tr>
+    <td class="sprk-u-FontWeight--bold">$body-font-src</td>
+    <td>string, url property</td>
+    <td>This tells Spark the url stack to use in downloading the web font for body level type.</td>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
    <td>This tells Spark the url stack to use in downloading the webfont for body level type.</td>
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  </tr>
+  <tr>
+    <td class="sprk-u-FontWeight--bold">$body-bold-font-src</td>
+    <td>string, url property</td>
+    <td>This tells Spark the url stack to use in downloading the web font for bold of body level type.</td>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
    <td>This tells Spark the url stack to use in downloading the webfont for bold of body level type.</td>
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  </tr>
+  <tr>
+    <td class="sprk-u-FontWeight--bold">$heading-font-src</td>
+    <td>string, url property</td>
+    <td>This tells Spark the url stack to use in downloading the web font for display level type.</td>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
    <td>This tells Spark the url stack to use in downloading the webfont for display level type.</td>
```

File: src/patterns/fundamentals/typography/setting-up.hbs [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  </tr>
+  <tr>
+    <td class="sprk-u-FontWeight--bold">$heading-bold-font-src</td>
+    <td>string, url property</td>
+    <td>This tells Spark the url stack to use in downloading the web font for bold of heading level type.</td>
```
Comment (afebbraro) [inline src/patterns/fundamentals/typography/setting-up.hbs]:
```suggestion
    <td>This tells Spark the url stack to use in downloading the webfont for bold of heading level type.</td>
```

File: packages/spark-core/base/_typography.scss [hunk 9]
```diff
@@ -1,6 +1,50 @@
 // ==================================================================
 // Typography
 // ==================================================================
+
+@if $using-webfont-body {
+  @font-face {
+    font-family: $body-font-name;
+    src: $body-font-src;
+    font-weight: normal;
```
Comment (afebbraro) [inline packages/spark-core/base/_typography.scss]: The weights should be set to vars.

File: packages/spark-core/base/_typography.scss [hunk 10]
```diff
@@ -1,6 +1,50 @@
...
+  }
+  @font-face {
+    font-family: $body-font-name;
+    src: $body-bold-font-src;
+    font-weight: bold;
```
Comment (afebbraro) [inline packages/spark-core/base/_typography.scss]: The weights should be set to vars.

File: packages/spark-core/base/_typography.scss [hunk 11]
```diff
@@ -1,6 +1,50 @@
...
+@if $using-webfont-heading {
+  @font-face {
+    font-family: $heading-font-name;
+    src: $heading-font-src;
+    font-weight: normal;
+  }
+  @font-face {
+    font-family: $heading-font-name;
+    src: $heading-bold-font-src;
+    font-weight: bold;
```
Comment (afebbraro) [inline packages/spark-core/base/_typography.scss]: The weights should be set to vars.

File: packages/spark-core/base/_typography.scss [hunk 12]
```diff
@@ -1,6 +1,50 @@
...
+    font-family: $body-font-name;
+    src: $body-bold-font-src;
+    font-weight: bold;
+  }
+}
+@if $using-webfont-heading {
+  @font-face {
+    font-family: $heading-font-name;
+    src: $heading-font-src;
+    font-weight: normal;
```
Comment (afebbraro) [inline packages/spark-core/base/_typography.scss]: The weights should be set to vars.

File: packages/spark-core/tests/typography.tests.js [hunk 13]
```diff
@@ -0,0 +1 @@
+// TODO: write tests!
```
Comment (afebbraro) [inline packages/spark-core/tests/typography.tests.js]: Forgot to do a TODO here.

File: packages/spark-core/base/typography.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
+/* global document Promise */
+const FontFaceObserver = require('fontfaceobserver');
+
+const typeLoader = ({ bodyFontName, headingFontName } = {}) => {
+  let bodyFont;
+  let headingFont;
+
+  if (bodyFontName) {
```
Comment (afebbraro) [inline packages/spark-core/base/typography.js]: We'll need to provide a useful error message if this fails.

Comment (RCopeland) [reply packages/spark-core/base/typography.js]: I think the useful error message issue is project-wide. I want to leave it as is until we decide how to handle it.

Comment (afebbraro) [reply packages/spark-core/base/typography.js]: It is is project-wide, but we gotta start somewhere.

Comment (RCopeland) [reply packages/spark-core/base/typography.js]: There's a lot of decisions to be made before we do error handling. I think it'd be wiser to create an issue to work it out before implementing.

Comment (RCopeland) [reply packages/spark-core/base/typography.js]: Created an issue to get it started.
https://github.com/sparkdesignsystem/spark-design-system/issues/687

**Pull request merged**